### DISCUSSION TITLE
Any way to remove something from autosuggest or clearing history?

### DISCUSSION THREAD

**Discussion by mboeker** [chrisant996/clink#672] [Q&A] [answered]
This is very low priority because it's cosmetic, since autosuggest goes with the most recently used command, but one time I accidentally typed 'exiot' instead of 'exit' and now it shows up every time. Is there a way to remove that suggestion? Clearing the history would work, but I tried turning history.save false and then true again and that did not clear out the history.

**Comment by chrisant996** [comment, marked as answer]
I recommend reading the documentation about history; there's a lot of useful information available there.

[Saved Command History](https://chrisant996.github.io/clink/clink.html#saved-command-history)

Key Binding | Description | Command Name
-- | -- | --
<kbd>Alt</kbd>-<kbd>Ctrl</kbd>-<kbd>D</kbd> | While navigating through the history, removes the current history line from the saved history. Otherwise has no effect. | [`remove-history`](https://chrisant996.github.io/clink/clink.html#rlcmd-remove-history)

So, for example you could:
- Use <kbd>Up</kbd> to navigate the history until the `exiot` line is recalled, then press <kbd>Alt</kbd>-<kbd>Ctrl</kbd>-<kbd>D</kbd> to remove that history entry.
- Or type `exi` then press <kbd>PgUp</kbd> until the `exiot` line is recalled, then press <kbd>Alt</kbd>-<kbd>Ctrl</kbd>-<kbd>D</kbd> to remove that history entry.
- Or run `history | findstr exiot` to find the history entry number associated with the command, and run <code>history delete <em>the_number_for_the_exiot_entry</em></code>.
- (Or several other variations on any of the above.)

P.S.  Also, run `history -h` for help on the history command.  The `history clear` command deletes the entire history.  But you don't need to do that.

**Comment by mboeker** [comment]
Oh I was focusing on autosuggest. Thank you for that detailed response!

**Reply by chrisant996** [reply]
[Auto-suggestions](https://chrisant996.github.io/clink/clink.html#gettingstarted_autosuggest) use multiple [strategies](https://chrisant996.github.io/clink/clink.html#autosuggest_strategy) to generate suggestions.

One strategy is "history" which pulls suggestions from history.  So, to make it stop suggesting a particular history entry, you would need to delete that history entry, so that is not there to be found anymore.
